core/src/SYCL/Kokkos_SYCL.cpp

```diff
@@ -59,6 +59,8 @@ SYCL::SYCL(const sycl::queue& stream)
         ptr->finalize();
         delete ptr;
       }) {
+  if (!stream.is_in_order())
+    Kokkos::abort("User provided sycl::queues must be in-order!");
```

Comment:
```suggestion
#ifdef KOKKOS_IMPL_SYCL_USE_IN_ORDER_QUEUES
  if (!stream.is_in_order())
    Kokkos::abort("User provided sycl::queues must be in-order!");
#endif
```

Reply:
I thought about it but decided to play it safe and make the check independent of what we do internally. What are your thoughts?